bundles/ch.elexis.stickynotes/src/ch/elexis/stickynotes/ui/StickyNotesView.java

```diff
@@ -134,19 +134,20 @@ private void deselect() {
 	private void doSelect(Patient pat) {
 		if (pat == null) {
 			deselect();
-
 		} else {
-
 			actPatient = pat;
 			actNote = StickyNote.load(actPatient);
 			etf.setText(actNote.getText());
-			// form.setText(actPatient.getLabel());
 			setPartName(Messages.StickyNotesView_StickyNotesNameDash + actPatient.getLabel());
-			RGB rgb = PreferenceConverter.getColor(prefs, Preferences.COLBACKGROUND);
-			UiDesk.getColorRegistry().put(Preferences.COLBACKGROUND, rgb);
+			RGB rgbBackground = PreferenceConverter.getColor(prefs, Preferences.COLBACKGROUND);
+			RGB rgbForeground = PreferenceConverter.getColor(prefs, Preferences.COLFOREGROUND);
+			if (rgbBackground.equals(rgbForeground)) {
+				rgbBackground = new RGB(255, 255, 0);
+				rgbForeground = new RGB(0, 0, 0);
```

Comment:
Don't do new RGB. This is a possible resource leak. Use `JFaceResources.getColorRegistry()` to initialize Colors or `	Display.getCurrent().getSystemColor(SWT.COLOR_BLACK)`. By doing a new here, you are responsible for disposing this. See https://www.eclipse.org/articles/swt-design-2/swt-design-2.html for more info on this